### PR TITLE
Allow consistent management of metrics related config settings from environment

### PR DESCRIPTION
**Pull request by gvelez17**
## Description

This will allow node operators to enable prometheus metrics as well as collector metrics from the node environment

## How Has This Been Tested?

Describe the tests that you ran to verify your changes. Provide instructions for reproduction.

- [ ] unit tests
- [x] manual testing with env settings on/off

## PR checklist

Before submitting this PR, please make sure:

- [ ] I have tagged the relevant reviewers and interested parties
- [x] I have updated the READMEs of affected packages
- [x] I have made corresponding changes to the documentation

## References:

Please list relevant documentation (e.g. tech specs, articles, related work etc.) relevant to this change, and note if the documentation has been updated.

### CODE REVIEW

**Comment by gvelez17** [comment]
Tested locally and verified that the environment variables result in the appropriate config.

File: docs-dev/ADVANCED_CONFIG.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+```
+{
+    "anchor": {
+        "anchor-service-url": STRING // the URL of the Ceramic Anchor Service
+        "auth-method": STRING        // the auth method used to access the Ceramic Anchor Service
+        "ethereum-rpc-url": STRING   // the RPC URL to query ethereum transactions
+    },
+    "http-api": {
+        "admin-dids": ARRAY               // array of string DIDs with access to Admin API
+        "cors-allowed-origins": [ ".*" ]  // allow all origins
```
Comment (3benbox) [inline docs-dev/ADVANCED_CONFIG.md:19]: This should be an ARRAY as well?

File: docs-dev/ADVANCED_CONFIG.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        "hostname": STRING                // hostname to bind and listen on
+        "port": INTEGER                   // port to listen on
+    },
+    "indexing": {
+        "allow-queries-before-historical-sync": false  // set to true to allow queries without sync
```
Comment (3benbox) [inline docs-dev/ADVANCED_CONFIG.md:24]: This should be BOOL?

File: docs-dev/ADVANCED_CONFIG.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+         "prometheus-exporter-port": NUMBER     // port for scraping prometheus metrics if enabled
+    },
+    "network": {
+        "name": STRING         // Network to connect to: mainnet, testnet-clay, or dev-unstable
+        "pubsub-topic": STRING // CAUTION- setting this will disconnect you from the main ceramic network
+    },
+    "node": {
+        "disable-composedb": BOOLEAN  // CAUTION if true, turns off composedb indexing
+        "private-seed-url": STRING    // the private seed for this node, or will be randomly generated
+        "readOnly": false             // run the node in read-only mode
```
Comment (3benbox) [inline docs-dev/ADVANCED_CONFIG.md:54]: Should be BOOLEAN too

File: docs-dev/ADVANCED_CONFIG.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        "stream-cache-limit": INTEGER // Max number of streams to keep in memory
+        "sync-override": STRING       // CAUTION overrides internal sync setting for stream loads
+    },
+    "state-store": { 
+				"local-directory": STRING // Defaults to $HOME/.ceramic/statestore
+        "mode": "fs|s3",          // volume storage option shared here, fs=>filesystem, s3=>s3 bucket
```
Comment (3benbox) [inline docs-dev/ADVANCED_CONFIG.md:60]: STRING with choice

File: docs-dev/ADVANCED_CONFIG.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+| Environment Variable | Config Section | Config Variable |
+|-----------------|-----------------|-----------------|
+| CERAMIC_DISABLE_IPFS_PEER_DATA_SYNC  | ipfs  | disable-peer-data-sync |
+| CERAMIC_INDEXING_DB_URI    | indexing    | db    |
+| CERAMIC_METRICS_EXPORTER_ENABLED  |  metrics   |  metrics-exporter-enabled   |
+| CERAMIC_NODE_PRIVATE_SEED_URL   | node    |  private-seed-url   |
+| CERAMIC_PROMETHEUS_EXPORTER_ENABLED | metrics  | prometheus-exporter-enabled |
+| CERAMIC_PROMETHEUS_EXPORTER_PORT   | metrics    | prometheus-exporter-port    |
+| COLLECTOR_HOSTNAME  | metrics    | collector-host    |
```
Comment (3benbox) [inline docs-dev/ADVANCED_CONFIG.md:80]: Just a comment, I'm kind of shocked there are so few, I thought we have an env for each config file element.

**Review by 3benbox — COMMENTED** [review]
Awesome stuff!

Just some consistency comments, looks great.

File: docs-dev/ADVANCED_CONFIG.md (hunk 5 above)
Comment (gvelez17) [reply docs-dev/ADVANCED_CONFIG.md:80]: ha yeah.  but is so

File: docs-dev/ADVANCED_CONFIG.md (hunk 3 above)
Comment (gvelez17) [reply docs-dev/ADVANCED_CONFIG.md]: Done

File: docs-dev/ADVANCED_CONFIG.md (hunk 4 above)
Comment (gvelez17) [reply docs-dev/ADVANCED_CONFIG.md]: Done

File: docs-dev/ADVANCED_CONFIG.md (hunk 2 above)
Comment (gvelez17) [reply docs-dev/ADVANCED_CONFIG.md]: Done

**Review by 3benbox — APPROVED** [review]
lgtm

**Pull request merged**